src/backend/distributed/deparser/citus_ruleutils.c

```diff
@@ -1039,7 +1039,12 @@ pg_get_indexclusterdef_string(Oid indexRelationId)
 										  0, 0, 0);
 	if (!HeapTupleIsValid(indexTuple))
 	{
-		ereport(ERROR, (errmsg("cache lookup failed for index %u", indexRelationId)));
+		if (EnablePropagationWarnings)
```

Comment:
`pg_get_indexclusterdef_string` is only called for `deparsing` queries. 

How a PG index comes here? We should most likely prevented the code flow coming here instead of masking an ERROR here?

Reply:
This is also unnecessary check. Tests pass without it.